### ISSUE TITLE
move obj.meta.datainfo to xPlt Constructor

### ISSUE THREAD

**Issue by erik-roberts** [davestanley/MDD#43]
since it's only used with xPlt, not nDDict. So moving it to xPlt. Might be best to remove it completely.

**Comment by erik-roberts** [comment]
8264de8

**Comment by erik-roberts** [comment]
@davestanley what do you think? should we keep it or remove it?

**Comment by benpolletta** [comment]
Eventually, we wanted to move matrix_dim_n axes (e.g. as created by
packDim) to meta.datainfo. I was thinking about making this an nDDictAxis
structure, perhaps with a different name, such as dataAxis or something.
What do you guys think?

**Comment by davestanley** [comment]
@benpolletta Yes, I think we should definitely make `matrix_dim_n` axis created by packDim2Mat (formerly packDim) an nDDictAxis.

As for the question of whether or not to move this into datainfo: in one sense, this would make datainfo something of a field for "internal use" by MDD. But if the user is writing to this as well, then this would mix functions for internal usage with user-editable functions. My initial instinct is to keep these two separate, but I'm open to other input here @erik-roberts  & Ben

**Comment by davestanley** [comment]
Also, in the future I would like to remove xPlt entirely and just merge both this and nDDict into MDD. This is because, as Ben has shown, plotRecursive can be used for other purposes besides just plotting.

**Comment by benpolletta** [comment]
Just a quick note of clarification: matrix_dim_n is already a scalar nDDictAxis. I was wondering whether we should make something like matrixAxis, a field which is a vector nDDictAxis. This would be easier to reorder when permuting matrix dimensions.

**Comment by erik-roberts** [comment]
can now close this then?
